formats/roam/roam.ts

```diff
@@ -0,0 +1,327 @@
+	blockText = blockText.replaceAll("{{[[DONE]]}}", "[x]");
+	blockText = blockText.replaceAll("{{word-count}}", "");
+	blockText = blockText.replaceAll("{{date}}", "");
+	blockText = blockText.replaceAll("{{[[POMO]]}}", "");
+	blockText = blockText.replaceAll("{{POMO}}", "");
+	blockText = blockText.replaceAll("{{slider}}", "");
+	blockText = blockText.replaceAll("{{[[slider]]}}", "");
+    blockText = blockText.replaceAll("{{encrypt}}", "");
+	blockText = blockText.replaceAll("{{[[encrypt]]}}", "");
+
+	blockText = blockText.replaceAll("{{TaoOfRoam}}", "");
+	blockText = blockText.replaceAll("{{orphans}}", "");
+	blockText = blockText.replaceAll("{{[[orphans]]}}", "");
+	blockText = blockText.replaceAll("{{count}}", "");
+	blockText = blockText.replaceAll("{{character-count}}", "");
+	blockText = blockText.replaceAll("{{comment-button}}", "");
+	blockText = blockText.replace("::", ":"); // Attributes::
+
+	blockText = blockText.replaceAll(/{{.*?\bvideo\b.*?(\bhttp.*?\byoutu.*?)}}/g, "![]($1)"); // youtube embeds
+	blockText = blockText.replaceAll(/(https?:\/\/twitter\.com\/(?:#!\/)?\w+\/status\/\d+(?:\?[\w=&-]+)?)/g, "![]($1)"); // twitter embeds
+	blockText = blockText.replaceAll(/{{.*?roam\/render[^}]*}}/g, ""); // {{roam/render}} components
+	blockText = blockText.replaceAll(/\_\_(.+?)\_\_/g, "*$1*"); // __ __ itallic
+	blockText = blockText.replaceAll(/\^\^(.+?)\^\^/g, "==$1=="); // ^^ ^^ highlight
+    
+	// block and page embeds
+	blockText = blockText.replaceAll(/{{\[{0,2}embed.*?(\(\(.*?\)\)).*?}}/g, "$1")
+	blockText = blockText.replaceAll(/{{\[{0,2}embed.*?(\[\[.*?\]\]).*?}}/g, "$1")
+	// download files uploaded to Roam
+	if (downloadAttachments) {
+		if (blockText.includes('firebasestorage')) {
+			// console.log(app)
+			blockText =  await downloadFirebaseFile(blockText, attachmentsFolder)
+		}
+	}
+    // blockText = blockText.replaceAll("{{[[table]]}}", ""); 
+	// blockText = blockText.replaceAll("{{[[kanban]]}}", "");
+	// blockText = blockText.replaceAll("{{mermaid}}", "");
+	// blockText = blockText.replaceAll("{{[[mermaid]]}}", "");
+    // blockText = blockText.replaceAll("{{diagram}}", "");
+	// blockText = blockText.replaceAll("{{[[diagram]]}}", "");
+
+	//   blockText = blockText.replaceAll(/\!\[(.+?)\]\((.+?)\)/g, "$1 $2"); //images with description
+	//   blockText = blockText.replaceAll(/\!\[\]\((.+?)\)/g, "$1"); //imags with no description
+	//   blockText = blockText.replaceAll(/\[(.+?)\]\((.+?)\)/g, "$1: $2"); //alias with description
+	//   blockText = blockText.replaceAll(/\[\]\((.+?)\)/g, "$1"); //alias with no description
+	//   blockText = blockText.replaceAll(/\[(.+?)\](?!\()(.+?)\)/g, "$1"); //alias with embeded block (Odd side effect of parser)
+
+	return blockText;
+  };
+
+async function jsonToMarkdown(graphFolder:string, attachmentsFolder:string, downloadAttachments:boolean,json: JsonObject, indent: string = '', isChild: boolean = false ): Promise<string> {
+    let markdown = '';
```

Comment:
Can we use an array and `join('')` at the end instead of keep concatenating the string?